### DISCUSSION TITLE
“conda install -c open3d-admin open3d” → not found

### DISCUSSION THREAD

**Discussion by cwreynolds** [isl-org/Open3D#6032] [Q&A] [answered]
I want to use Open3D from Python. I've never touched Open3D, I know a bit of Python, my experience with Anaconda is “used it once, got it working, never touched it again.” I found https://anaconda.org/open3d-admin/open3d which suggestes: `conda install -c open3d-admin open3d`. On my macOS 12.6.3 M1 laptop, I did:

    conda create -n open3d_2023 python=3.10
    conda activate open3d_2023
    conda install -c open3d-admin open3d

The install got “not found” while mentioning that current channels include https://conda.anaconda.org/open3d-admin/osx-arm64

How should I proceed? The Open3D instructions on GitHub suggest `pip install open3d`. I do not understand the interaction between conda and the older(?)/underlying(?) pip. Can I just run the pip command while the conda environment is active?

**Comment by cwreynolds** [comment]
Hoping that the virtual environment would contain any fallout, I decided to just try it. The pip worked, but the installation test code snippet (from https://github.com/isl-org/Open3D) failed.

    pip install open3d

> Successfully installed Flask-2.2.3 Jinja2-3.1.2 MarkupSafe-2.1.2 addict-2.4.0 attrs-22.2.0 click-8.1.3 configargparse-1.5.3 contourpy-1.0.7 cycler-0.11.0 dash-2.9.1 dash-core-components-2.0.0 dash-html-components-2.0.0 dash-table-5.0.0 fastjsonschema-2.16.3 fonttools-4.39.2 itsdangerous-2.1.2 joblib-1.2.0 jsonschema-4.17.3 jupyter-core-5.3.0 kiwisolver-1.4.4 matplotlib-3.7.1 nbformat-5.7.0 numpy-1.24.2 open3d-0.17.0 packaging-23.0 pandas-1.5.3 pillow-9.4.0 platformdirs-3.2.0 plotly-5.13.1 pyparsing-3.0.9 pyquaternion-0.9.9 pyrsistent-0.19.3 python-dateutil-2.8.2 pytz-2023.2 pyyaml-6.0 scikit-learn-1.2.2 scipy-1.10.1 six-1.16.0 tenacity-8.2.2 threadpoolctl-3.1.0 tqdm-4.65.0 traitlets-5.9.0 werkzeug-2.2.3

    python -c "import open3d as o3d; print(o3d.__version__)"
> Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/cwr/miniforge3/envs/open3d_2023/lib/python3.10/site-packages/open3d/__init__.py", line 93, in <module>
    from open3d.cpu.pybind import (core, camera, data, geometry, io, pipelines,
ImportError: dlopen(/Users/cwr/miniforge3/envs/open3d_2023/lib/python3.10/site-packages/open3d/cpu/pybind.cpython-310-darwin.so, 0x0002): Library not loaded: '/opt/homebrew/opt/libomp/lib/libomp.dylib'
  Referenced from: '/Users/cwr/miniforge3/envs/open3d_2023/lib/python3.10/site-packages/open3d/cpu/pybind.cpython-310-darwin.so'
  Reason: tried: '/opt/homebrew/opt/libomp/lib/libomp.dylib' (no such file), '/usr/local/lib/libomp.dylib' (no such file), '/usr/lib/libomp.dylib' (no such file)

**Comment by cwreynolds** [comment, marked as answer]
WOT!? Nevermind, problem solved. Sorry for the “talking to myself” posts. Taking a moment to read the error above, realizing that it was looking for something called `libomp`, looked it up, did a:

    brew install libomp

Then everything **Just Worked**. Was able to run all three suggested tests: Verify installation, Python API, and Open3D CLI.

Thanks for Open3D. It looks amazing!

**Comment by yalov** [comment]
as of now, the last version of open3d-admin open3d on any x64 system is 0.15.1, 
and it depends on python 3.8.

I was unable to find conda package of later versions, that able to be installed on python 3.10, 
so I am not sure how you manage it.

**Comment by mdrahmed** [comment]
I am trying to use open3d in linux. I don't think, python-3.10 is compatible with open3d deps. So, I did this and it worked fine,
```
conda create -n open3d_env python=3.8
conda activate open3d_env
conda install -c open3d-admin open3d
```
Also, ensure that the open3d-admin channel is correctly configured in your Conda channels,
```
conda config --show channels
conda config --add channels open3d-admin
```

**Comment by bardia-ardakanian** [comment]
Hello everyone, I know I am late to the party but here is what I have found out:

if you want to use both ```PyTorch``` and ```open3d```, you should install open3d first because for some versions of ```torch``` (in my case ```torch==1.12.1```) the ```open3d``` will not load and result in segment fault. I'm entirely not sure why this is the case but ```conda``` fails to install these packages together and you always should use '''pip''' to install one of them.

some notes on this matter:
- This method only works for ```python==3.8```. It doesn't work for ```python=3.9, 3.10, 3.11```
- For installing either ```open3d``` or ```torch``` you have to use ```pip```. Using ```conda``` to install them both will result in dependency incompatibility.

```
conda create -n myenv python=3.8
conda activate myenv
conda install -c open3d-admin open3d
pip install torch==1.12.1 torchvision==0.13.1 torchaudio==0.12.1 
```

hope this helps! many thanks to @mdrahmed.
